python/lsst/ts/watcher/alarm.py

```diff
@@ -0,0 +1,205 @@
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["Alarm"]
+
+import time
+
+from lsst.ts.idl.enums.Watcher import AlarmSeverity
+from lsst.ts import salobj
+
+
+class Alarm:
+    """A Watcher alarm.
+
+    Parameters
+    ----------
+    name : `str`
+        Name of alarm. This must be unique among all alarms
+        and should be of the form system_[subsystem_...]_name
+        so that groups of related alarms can be acknowledged.
+        Use underscores instead of dots to make the regular expressions
+        or glob expressions easier to write.
+    callback : ``callable``
+        Function or coroutine to call whenever the alarm changes state.
+    """
+    def __init__(self, name, callback):
+        self.name = name
+        self.callback = callback
+
+        self.severity = AlarmSeverity.NONE
+        self.max_severity = AlarmSeverity.NONE
+        self.reason = ""
+        self.acknowledged = False
+        self.acknowledged_by = ""
+        self.enabled = True
+
+        self.timestamp_severity_oldest = 0
+        self.timestamp_severity_newest = 0
+        self.timestamp_max_severity = 0
+        self.timestamp_acknowledged = 0
+
+    def __eq__(self, other):
+        for field in ("name",
+                      "callback",
+                      "severity",
+                      "max_severity",
+                      "reason",
+                      "acknowledged",
+                      "acknowledged_by",
+                      "enabled",
+                      "timestamp_severity_oldest",
+                      "timestamp_severity_newest",
+                      "timestamp_max_severity",
+                      "timestamp_acknowledged"):
+            if getattr(self, field) != getattr(other, field):
+                return False
+        return True
+
+    def __ne__(self, other):
+        return not self.__eq__(other)
+
+    def set_severity(self, severity, reason):
+        """Set the severity.
+
+        Parameters
+        ----------
+        severity : `AlarmSeverity` or `int`
+            New severity.
+        reason : `str`
+            The reason for this state; this should be a brief message
+            explaining what is wrong. Ignored if severity is NONE.
+
+        Returns
+        -------
+        updated : `bool`
+            True if the alarm state changed (i.e. if any fields were modified),
+            False otherwise.
+        """
+        severity = AlarmSeverity(severity)
+        if severity == AlarmSeverity.NONE and self.quiescent:
```

Comment:
quiescent: I had to look this one up ;-)

Reply:
That's not a good sign. If you found it confusing then others might also find it confusing. I am open to other suggestions.

Reply:
Maybe dormant or inactive?

Reply:
`inactive` sounds like the opposite of `active`, which this is not. Also I don't like testing for a negative name as `not inactive` is difficult to parse. Dormant might work. I'll ask on the DM naming channel. I confess I slightly prefer `quiescent` as I think it a bit more accurate.

Reply:
And the winner is...."nominal".